### ISSUE TITLE
Cannot reproduce the results with the provided weights

### ISSUE THREAD

**Issue by ntlm1686** [CAMMA-public/rendezvous#26]
With the latest version of ivtmetrics and sci-kit learn, and [ Rendezvous | ResNet-18 | Low | CholecT50 | RDV ] model, I am getting a much lower mAP.
mAP I: 0.7156816255213027
mAP V: 0.43615357074871014
mAP T: 0.2512772328168727

Am I using the correct weight? Or is this the problem with the sci-kit learn version?

Maybe this is due to this issue: https://github.com/CAMMA-public/ivtmetrics/issues/3.

**Comment by nwoyecid** [comment]
Please ensure you do not update your sckit learn installing ivtmetrics so that it will use its compiled version.

**Comment by ntlm1686** [comment]
@nwoyecid 

This is [my evaluation script](https://gist.github.com/ntlm1686/058144198cca42daea2ce602821a001f), I have tried `ivtmetrics` and my own implementation, neither do they reproduce the results in the paper.

As the cholect50 data has been updated, it doesn't contain the `.txt` files for annotations. Could you please provide a new evaluation script that is compatible with the latest data format that can reproduce the results?

**Comment by ntlm1686** [comment]
@nwoyecid 
Could you please also provide the required ivtmetrics version and scikit-learn version?

**Comment by labdeeman7** [comment]
Were you able to solve the evaluation issue @ntlm1686 ?

**Comment by labdeeman7** [comment]
Here’s what worked for me — hope it helps!

### IVT Recognition (ivtrecognition)

1. Library Versions Matter: Make sure you're using older versions of scikit-learn (1.0.2 worked for me) and  numpy<2.0 (I used 1.26.1). 
Older sklearn produces NaNs in some situations where the newer sklearn produces 0 (has to do with TP and FP). You want the NaNs. 

2. Pass Logits, Not One-Hot Outputs
The recognition metric expects raw logits or confidence scores — not one-hot outputs.

3. Use compute_video_AP()
Don't use compute_AP() or compute_global_AP() for recognition tasks.
compute_video_AP() is the correct method, and it: Requires frames to be passed sequentially. Needs you to explicitly signal the end of each video using video_end(). 


###  IVT Detection (ivtdetection)
No Sklearn. So you dont have to worry about versioning.  
For detection metrics, pass: instrument_id, triplet_id, confidence score. Do not pass the full logits here — this behaves more like traditional object detection metrics.


### Final Notes
Even though ivtrecognition and ivtdetection look like they share the same interface, they evaluate very different things:
ivtrecognition: evaluates classification via Average Precision (e.g., action/target prediction quality).
ivtdetection: evaluates object detection-like performance (AP over triplet detections).

**Comment by nwoyecid** [comment]
This issue has been previously resolved

**Comment by nwoyecid** [comment]
This is now closed.

**Comment by WangShimei** [comment]
> Were you able to solve the evaluation issue [@ntlm1686](https://github.com/ntlm1686) ?

Hello, I meet the same problem, can you help me?
I’m running into an issue reproducing the reported results with the Rendezvous checkpoint [Rendezvous | ResNet-18 | Low | CholecT50 | Challenge].

Command:
python run.py -e --dataset_variant cholect50-challenge --test_ckpy rendezvous/pytorch/checkpoint/rendezvous_l8_cholect50_challenge_k0_batchnorm_lowres.pth

However, the evaluation results are much lower than expected:

Mean AP (I | V | T | IV | IT | IVT):
0.7786 | 0.4637 | 0.2936 | 0.2174 | 0.1611 | 0.0919

Could you please advise what might be wrong? Thanks a lot!

**Comment by gezishifanfan** [comment]
@labdeeman7 It works! Thanks a lot.
